### PR TITLE
Support for Postgres as configuration building block

### PR DESCRIPTION
**Pull request by akhilac1**
Signed-off-by: akhilac1 <chetlapalle.akhila@gmail.com>

# Description
Added postgres as a configuration store in dapr. 
Issue - #4551 
Contrib code is merged - https://github.com/dapr/components-contrib/pull/2063
docs PR has been created, review is pending - https://github.com/dapr/docs/issues/2779

Added a new file in dapr/cmd/daprd/components/configuration_postgres.go to register postgres as a supported configuration store
## Issue reference  -- #4551 

Please reference the issue this PR will close: #4551 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [X] Unit tests passing
* [ ] End-to-end tests passing
* [X] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#https://github.com/dapr/docs/issues/2779
* [X] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs - https://github.com/dapr/docs/issues/2779
* [X] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#https://github.com/dapr/docs/issues/2779

### CODE REVIEW

**Comment by akhilac1** [comment]
@artursouza - created a new PR due to too many merge conflicts in #5227

File: go.mod [hunk 1]
```diff
@@ -396,6 +398,7 @@ replace github.com/eclipse/paho.mqtt.golang => github.com/shivamkm07/paho.mqtt.g
 // Don't commit with this uncommented!
 //
 // replace github.com/dapr/components-contrib => ../components-contrib
+
```
Comment (ItalyPaleAle) [inline go.mod:401]: Please don't change this line

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

**Comment by ItalyPaleAle** [comment]
Need to re-run `make modtidy-all`

**Review by artursouza — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

**Comment by akhilac1** [comment]
@artursouza - this PR is ready to be merged.

**Review by ItalyPaleAle — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**